firecrown/likelihood/likelihood.py

```diff
@@ -83,15 +83,31 @@ def __init__(self, parameter_prefix: Optional[str] = None) -> None:
         """Default initialization for a base Likelihood object."""
         super().__init__(parameter_prefix=parameter_prefix)
 
-        self.predicted_data_vector: Optional[npt.NDArray[np.double]] = None
-        self.measured_data_vector: Optional[npt.NDArray[np.double]] = None
-        self.inv_cov: Optional[npt.NDArray[np.double]] = None
-        self.statistics: UpdatableCollection = UpdatableCollection()
-
     @abstractmethod
     def read(self, sacc_data: sacc.Sacc) -> None:
         """Read the covariance matrix for this likelihood from the SACC file."""
 
+    def make_realization_vector(self) -> npt.NDArray[np.float64]:
+        """Create a new realization of the model using the previously computed
+        theory vector and covariance matrix.
+        """
+        raise NotImplementedError(
+            "This class does not implement make_realization_vector."
+        )
+
+    def make_realization(
```

Comment:
`mypy` misses that this method does not return the required `sacc.Sacc` object.
This method should raise a `NotImplementedError`, so that it is only an error to use a subclass that lacks this method if that method is called.